### PR TITLE
feat(tabs): tabs have a badge

### PR DESCRIPTION
**Pull request by plowsirius**
ref #1536 

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1601)

### CODE REVIEW

File: components/tabs/Badge.web.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
```
Comment (paranoidjk) [inline components/tabs/Badge.web.tsx]: 为什么要重写一遍 Badge？ 没法复用 Badge 组件吗？

File: components/tabs/index.web.tsx [hunk 2]
```diff
@@ -47,12 +48,34 @@ export default class Tabs extends React.Component<TabsProps, any> {
...
+    const children: any[] = [];
+    React.Children.forEach(this.props.children, (c: any) => {
+      children.push(c);
+    });
+    const panels = children.map((c: any) => {
+      const cProps = c.props;
+      const tab = (<Tab
+        tab={cProps.tab}
+        prefixCls={`${this.props.prefixCls}-tabpane`}
+        badge={cProps.badge}
```
Comment (paranoidjk) [inline components/tabs/index.web.tsx]: 这个倾入性还是太强了，既然 `TabPane.props.tab` 支持`React.Node`, https://github.com/react-component/tabs#tabpane 这个需求在用户端也是可以做的

另外 @silentcloud 说的按需加载我同意，大部分用户的 Tab 不需要 Badge，那 Badge 最好做成可插拔的功能，比如伪代码：

```tsx
import { Tabs, WhiteSpace, TabWithBadge } from 'antd-mobile';

const TabPane = Tabs.TabPane;

const TabExample = () => (
  <div>
    <Tabs defaultActiveKey="1" >
      <TabPane tab={<TabWithBadge badge={3}> title </TabWithBadge>} key="1">
        <div >
          Content of First Tab
        </div>
      </TabPane>
      <TabPane tab="Second Tab" key="2" >
        <div >
          Content of Second Tab
        </div>
      </TabPane>
    </Tabs>
    <WhiteSpace />
  </div>
);

ReactDOM.render(<TabExample />, mountNode);
```

**Comment by paranoidjk** [comment]
@pingan1927 

体积问题成为 antd-mobile 在 C 端使用的阻碍，这也是来自你们业务线的反馈。

我们要往轻量化和按需加载走，不是 all in one，就算牺牲一点点 API 的易用性 ref  https://github.com/ant-design/ant-design-mobile/issues/1176

我的建议是 **低频次的 feature** 最好实现为按需加载的。 ref https://github.com/ant-design/ant-design-mobile/issues/777

**Comment by pingan1927** [comment]
如果有更好的方案就按最合理的方式解决。keypoint是满足业务的前提下做尽可能的轻量化。

**Comment by pingan1927** [comment]
试了伪代码，可以满足业务需要，单需要更新下样式， @plowsirius 会补充下例子以及更新样式

**Comment by paranoidjk** [comment]
close since moved to https://github.com/ant-design/ant-design-mobile/pull/1601

**Pull request closed without merging**